frontend/src/pages/modelServing/screens/projects/NIMServiceModal/DeployNIMServiceModal.tsx

```diff
@@ -0,0 +1,327 @@
+import { DataConnection, NamespaceApplicationCase } from '~/pages/projects/types';
+import { getDisplayNameFromK8sResource, translateDisplayNameForK8s } from '~/concepts/k8s/utils';
+import { useAccessReview } from '~/api';
+import { SupportedArea, useIsAreaAvailable } from '~/concepts/areas';
+import KServeAutoscalerReplicaSection
+  from '~/pages/modelServing/screens/projects/kServeModal/KServeAutoscalerReplicaSection';
+import useGenericObjectState from '~/utilities/useGenericObjectState';
+import {
+  AcceleratorProfileSelectFieldState,
+} from '~/pages/notebookController/screens/server/AcceleratorProfileSelectField';
+import NIMPVCSizeSection from '~/pages/modelServing/screens/projects/NIMServiceModal/NIMPVCSizeSection';
+import { useDashboardNamespace } from '~/redux/selectors';
+
+const NIM_SECRET_NAME = 'nvidia-nim-secrets';
+const NIM_NGC_SECRET_NAME = 'ngc-secret';
+const NIM_PVC_NAME = 'nim-pvc';
+
+const accessReviewResource: AccessReviewResourceAttributes = {
+  group: 'rbac.authorization.k8s.io',
+  resource: 'rolebindings',
+  verb: 'create',
+};
+
+type DeployNIMServiceModalProps = {
+  isOpen: boolean;
+  onClose: (submit: boolean) => void;
+  servingRuntimeTemplates?: TemplateKind[];
+} & EitherOrNone<
+  {
+    projectContext?: {
+      currentProject: ProjectKind;
+      dataConnections: DataConnection[];
+    };
+  },
+  {
+    editInfo?: {
+      servingRuntimeEditInfo?: ServingRuntimeEditInfo;
+      inferenceServiceEditInfo?: InferenceServiceKind;
+      secrets?: SecretKind[];
+    };
+  }
+>;
+
+const DeployNIMServiceModal: React.FC<DeployNIMServiceModalProps> = ({
+  isOpen,
+  onClose,
+  servingRuntimeTemplates,
+  projectContext,
+  editInfo,
+}) => {
+  const [createDataServingRuntime, setCreateDataServingRuntime, resetDataServingRuntime, sizes] =
+    useCreateServingRuntimeObject(editInfo?.servingRuntimeEditInfo);
+  const [createDataInferenceService, setCreateDataInferenceService, resetDataInferenceService] =
+    useCreateInferenceServiceObject(
+      editInfo?.inferenceServiceEditInfo,
+      editInfo?.servingRuntimeEditInfo?.servingRuntime,
+      editInfo?.secrets,
+    );
+
+  const isAuthorinoEnabled = useIsAreaAvailable(SupportedArea.K_SERVE_AUTH).status;
+  const currentProjectName = projectContext?.currentProject.metadata.name;
+  const namespace = currentProjectName || createDataInferenceService.project;
+  const isInferenceServiceNameWithinLimit =
+    translateDisplayNameForK8s(createDataInferenceService.name).length <= 253;
+
+  const acceleratorProfileState = useServingAcceleratorProfile(
+    editInfo?.servingRuntimeEditInfo?.servingRuntime,
+    editInfo?.inferenceServiceEditInfo,
+  );
+  const [
+    selectedAcceleratorProfile,
+    setSelectedAcceleratorProfile,
+    resetSelectedAcceleratorProfile,
+  ] = useGenericObjectState<AcceleratorProfileSelectFieldState>({
+    profile: undefined,
+    count: 0,
+    useExistingSettings: false,
+  });
```

Comment:
I'm assuming this is where it was copied from: https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/pages/modelServing/screens/projects/kServeModal/ManageKServeModal.tsx#L129. Is it correct that the selectedAcceleratorProfile value is sourced from the AcceleratorProfileSelectFieldState object rather than the useAcceleratorProfileState hook?

Reply:
Nope, sorry -- they changed the implementation on me -- that's my bad. Continue with what you have.